packages/terra-navigation-prompt/docs/README.md

```diff
@@ -0,0 +1,138 @@
+
+The most straightforward solution is to lift the state up. If the state is lifted into a parent component, it can be more directly managed by the component responsible for navigation, or it could be persisted indefinitely. **This is the strategy that should be implemented by most components.** However, for very large applications or highly composable applications built out of many stateful components, lifting the state may not be feasible.
+
+Another solution is to expose callbacks from the stateful components. Whenever the stateful component's state changes, the callbacks are executed to notify the consumer of the current state. This works well at a small scale, but this becomes hard to maintain as the component tree increases in depth. One would have to chain these callbacks many levels deep to ensure every component is kept abreast of the current state, and every component, even those without these state concerns, would have to adhere to a consistent prop interface for those callbacks.
+
+The NavigationPrompt and the NavigationPromptCheckpoint were created smooth over the rough edges of the aforementioned strategies. Components that have sensitive state can render a NavigationPrompt to make their state known to the application. Components that navigate between components can use the NavigationPromptCheckpoint to check for the presence of NavigationPrompts and inform the user in a consistent way. State remains localized to the component that needs it, communication of state changes occurs automatically, and components that don't share these concerns can safely ignore all of it.
+
+## Usage
+
+### `NavigationPrompt`
+
+When a component renders a NavigationPrompt, it is telling its parent components two things: that the component is in the middle of something important, and that users may care about the component going away.
+
+The NavigationPrompt can be rendered with two props: a `description` string that gives context to the NavigationPrompt, and a `metaData` object that can be used to provide additional data that may be relevant to the NavigationPrompt.
+
+The NavigationPrompt can be rendered like any other component, but prop consistency should be maintained between renders when possible. Creating a new object for the `metaData` prop within a render function will cause the NavigationPrompt to deregister and reregister itself during every update:
+
+```jsx
+render() {
+  return (
+    <NavigationPrompt description="This prompt will register and notify every time the component renders" metaData={{ important: 'info' }} />
+  );
+}
+```
+
+To avoid this, the `metaData` value can be stored inside a component's state or an internal property and updated only when necessary:
+
+```jsx
```

Comment:
It'd be nice to update this to use diff syntax to quick see the changes you suggested

Reply:
```suggestion
```diff
+ constructor() {
+   this.state = {
+     metaDataValue: {
+       important: 'info',
+     }
+   }
+ }

render() {
  return (
    <NavigationPrompt
        description="My props are consistent between renders, so this prompt will not re-register or notify" 
-       metaData={{ important: 'info' }}
+       metaData={this.state.metaDataValue}
     />
  );
}

```